src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
+                    D_8088BFC0 += 0.01f;
+                } else {
+                    D_8088BFC0 -= 0.01f;
+                }
+            }
+
+            this->dyna.actor.speedXZ = this->dyna.actor.speedXZ + 0.05f;
+            this->dyna.actor.speedXZ = (this->dyna.actor.speedXZ > 2.0f) ? 2.0f : this->dyna.actor.speedXZ;
+
+            if (D_8088BFC0 > 0.0f) {
+                temp_v1 = Math_StepToF(&D_8088BFC0, 20.0f, this->dyna.actor.speedXZ);
+            } else {
+                temp_v1 = Math_StepToF(&D_8088BFC0, -20.0f, this->dyna.actor.speedXZ);
+            }
+
+            this->dyna.actor.world.pos.x = (Math_SinS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.x;
+            this->dyna.actor.world.pos.z = (Math_CosS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.z;
+
+            if (temp_v1 != 0) {
+                player->stateFlags2 &= ~0x10;
+                this->dyna.unk_150 = 0.0f;
+                this->dyna.actor.home.pos.x = this->dyna.actor.world.pos.x;
+                this->dyna.actor.home.pos.z = this->dyna.actor.world.pos.z;
+                D_8088BFC0 = 0.0f;
+                this->dyna.actor.speedXZ = 0.0f;
+                this->unk_16A = 5;
+            }
+
+            func_8002F974(&this->dyna.actor, NA_SE_EV_ROCK_SLIDE - SFX_FLAG);
+        } else {
+            player->stateFlags2 &= ~0x10;
+            this->dyna.unk_150 = 0.0f;
+            if (this->unk_16A != 0) {
+                this->unk_16A--;
+            }
+        }
+    } else {
+        this->unk_16A = 0;
+    }
+
+    sp2C = this->dyna.actor.world.pos.z - D_8088BF60.z;
+    if (sp2C < 0.5f) {
+        Flags_SetSwitch(globalCtx, this->dyna.actor.params);
+        Math_Vec3f_Copy(&this->dyna.actor.home.pos, &D_8088BF60);
+        this->dyna.actor.world.pos.x = D_8088BF60.x;
+        this->dyna.actor.world.pos.z = D_8088BF60.z;
+        this->dyna.actor.speedXZ = 0.0f;
+        D_8088BFC0 = 0.0f;
+        player->stateFlags2 &= ~0x10;
+        this->actionFunc = func_8088B79C;
+    }
+
+    frame = globalCtx->gameplayFrames & 0xFF;
+    if ((globalCtx->gameplayFrames & 0x100) != 0) {
+        this->unk_16C = 0.0f;
+    } else if (frame < 0x80) {
+        this->unk_16C = sinf(frame * (4 * 0.001f * M_PI)) * 19.625f;
+    } else if (frame < 0xE6) {
+        this->unk_16C = 19.625f;
+    } else {
+        this->unk_16C -= 1.0f;
+        this->unk_16C = (this->unk_16C < 0.0f) ? 0.0f : this->unk_16C;
+    }
+
+    if (sp2C < 100.0f) {
+        this->unk_16C = (this->unk_16C > 6.125f) ? 6.125f : this->unk_16C;
+    }
+}
+
+void func_8088B5F4(BgHidanRock* this, GlobalContext* globalCtx) {
+    s16 unk_16A;
+
+    unk_16A = this->unk_16A;
+    if (unk_16A != 0) {
+        this->unk_16A = unk_16A - 1;
+        unk_16A = this->unk_16A;
+    }
+
+    if (unk_16A == 0) {
+        func_8088B24C(this);
+    }
+}
+
+void func_8088B634(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (func_8004356C(&this->dyna) != 0) {
```

Comment:
```suggestion
    if (func_8004356C(&this->dyna)) {
```
I think this function is boolean.